File: notebooklm-import-raw/index_content.py
import os
from pathlib import Path

directory = Path('notebooklm-import-raw')
index = []

for file_path in directory.glob('*.txt'):
    if file_path.name == 'index.txt' or file_path.name == 'index_content.py':
        continue
    with open(file_path, 'r', encoding='utf-8', errors='ignore') as f:
        content = f.read()
        lines = content.split('\n')
        # Try to find a title
        title = lines[0].strip() if lines and lines[0].strip() else file_path.name
        # Look for abstract or introduction
        summary = ''
        if 'Abstract' in content:
            start = content.find('Abstract')
            end = content.find('\n\n', start)
            if end == -1:
                end = start + 1000
            summary = content[start:end].strip()
        elif 'Introduction' in content:
            start = content.find('Introduction')
            end = content.find('\n\n', start)
            if end == -1:
                end = start + 1000
            summary = content[start:end].strip()
        else:
            # First few lines
            summary = ' '.join([line.strip() for line in lines[1:10] if line.strip()])[:1000]
        index.append(f"File: {file_path.name}\nTitle: {title}\nSummary: {summary}\n\n")

with open(directory / 'content_index.txt', 'w', encoding='utf-8') as f:
    f.writelines(index)

File: database/raw/index_content.js
const fs = require('fs');
const path = require('path');

const directory = 'notebooklm-import-raw';
const files = fs.readdirSync(directory).filter(f => f.endsWith('.txt') && f !== 'index.txt' && f !== 'index_content.js');

let index = '';

for (const file of files) {
    const filePath = path.join(directory, file);
    const content = fs.readFileSync(filePath, 'utf8');
    const lines = content.split('\n');
    const title = lines[0].trim() || file;
    let summary = '';
    if (content.includes('Abstract')) {
        const start = content.indexOf('Abstract');
        const end = content.indexOf('\n\n', start);
        summary = content.substring(start, end !== -1 ? end : start + 1000);
    } else if (content.includes('Introduction')) {
        const start = content.indexOf('Introduction');
        const end = content.indexOf('\n\n', start);
        summary = content.substring(start, end !== -1 ? end : start + 1000);
    } else {
        summary = lines.slice(1, 10).join(' ').substring(0, 1000);
    }
    index += `File: ${file}\nTitle: ${title}\nSummary: ${summary.trim()}\n\n`;
}

fs.writeFileSync(path.join(directory, 'content_index.txt'), index);
